x/wasm/keeper/query_plugins.go

```diff
@@ -263,8 +264,19 @@ func IBCQuerier(wasm contractMetaDataSource, channelKeeper types.ChannelKeeper)
 	}
 }
 
+var queryDenyList = []string{
+	"/cosmos.tx",
+	"/cosmos.base.tendermint",
+}
+
 func StargateQuerier(queryRouter GRPCQueryRouter) func(ctx sdk.Context, request *wasmvmtypes.StargateQuery) ([]byte, error) {
 	return func(ctx sdk.Context, msg *wasmvmtypes.StargateQuery) ([]byte, error) {
+		for _, b := range queryDenyList {
+			if strings.Contains(msg.Path, b) {
+				return nil, wasmvmtypes.UnsupportedRequest{Kind: fmt.Sprintf("'%s' path is not allowed from the contract", msg.Path)}
```

Comment:
I want to be more sensible to not return user input in errors. Would 'path is not allowed from the contract' be good enough? The caller knows the path already.

Reply:
I updated this. The caller (user) will not know by default, as this is called from a contract. But I did redact.

Also, 3 lines lower we already had this code:
`return nil, wasmvmtypes.UnsupportedRequest{Kind: fmt.Sprintf("No route to query '%s'", msg.Path)}`

Which I wrote March 2021. Should we change that as well?